src/DataStructures/VectorImpl.hpp

```diff
@@ -199,6 +200,22 @@ class VectorImpl
     std::fill(data(), data() + set_size, value);
   }
 
+  /// Create from a copy of the given container
+  ///
+  /// \param container A container with a `value_type` that is the same as `T`.
+  /// Currently restricted to `std::vector<T>` and `std::array<T>`.
+  template <typename Container, typename U = typename Container::value_type,
```

Comment:
Why not inline the `Container::value_type` on the next line?

Reply:
done